### ISSUE TITLE
Too long time

### ISSUE THREAD

**Issue by moadbahron** [emilbjornson/competitive-cell-free#1]
Hi ,

The # of realizations when its 1000 , the code take a lot of time to get results, while the Matlab CPU is ~50% and RAM ~1.6 GB which is normal , but when I change the #of setups and #of realizations 20,100 respectively , the results are shown , but the curves are not smooth as your paper results .
![untitled1](https://user-images.githubusercontent.com/111454149/185205433-d54bd6b2-dd75-4812-9f66-b4a2993e6f9b.png)

Is my laptop is not suitable for your code or what it seems the issue?

**Comment by emilbjornson** [comment]
Hi!

You need a sufficient number of random data points on the curves to get smooth results from a Monte Carlo simulation. The number of points equals the number of scenarios. We used 200 points to get relatively smooth curves. 20 points as in your case is not enough.

I think you can get away with 100 realizations per scenario but you need 200 scenarios. The run time should be proportional to the number of scenarios. So 200 takes 10 times more time than 20.

It can take a few days to generate highly accurate simulation results.

**Comment by moadbahron** [comment]
Hi,

"Number of setups with random UE locations" in other words Is it the # of trails of random positions of UE location .
and 200 is the min or optimum value  for the curves to be smooth .

**Comment by emilbjornson** [comment]
Yes, we concluded that 200 setups was sufficient to get _visually_ smooth curves, but more setups will always lead to even smoother results so it is an accuracy vs. complexity tradeoff.

Even if the curves look smooth, there will be an error margin that one must take into consideration when comparing closely spaced curves. We are using the Monte Carlo method to create an empirical CDF and there are ways to characterize the estimation errors if you want to look deeper into it:
https://stats.stackexchange.com/questions/342315/accuracy-of-empirical-cumulative-distribution-function

**Comment by moadbahron** [comment]
Hi again ,

Right now I have running other disktops to run each Fig at your recommendation , But in the meantime I have selected 
#setups=20
#realizations =100 
on my laptop

The results are the followings,


![Fig2a_2](https://user-images.githubusercontent.com/111454149/185478193-c013c950-fcdd-4ed8-9f27-9b18607185af.png)
![Fig2a-1](https://user-images.githubusercontent.com/111454149/185478199-fae43d03-090c-4b4d-8f56-194dee313f8d.png)
![Fig2b](https://user-images.githubusercontent.com/111454149/185478200-e85fddd4-1250-4303-acad-f24afceeec47.png)
![Fig4-1](https://user-images.githubusercontent.com/111454149/185478202-ab05202b-8f85-47a7-8728-62a47e317e16.png)
![Fig4-2](https://user-images.githubusercontent.com/111454149/185478204-48706b07-9cdf-46e8-82d1-acfab58f2c97.png)
![Fig5](https://user-images.githubusercontent.com/111454149/185478207-33e8af96-f777-4ca4-9c94-9fd480532d06.png)
![Fig6](https://user-images.githubusercontent.com/111454149/185478209-2d1a5338-093a-4adf-b7ae-9292020e9541.png)
